ts/multicolor_series.ts

```diff
@@ -0,0 +1,729 @@
+                stroke: colorType,
+                'stroke-width': lineWidth,
+                fill: 'none',
+                zIndex: 1 // #1069
+            };
+
+            let item;
+
+            if (dashStyle) {
+                attribs.dashstyle = dashStyle;
+            } else if (roundCap) {
+                attribs['stroke-linecap'] =
+                    attribs['stroke-linejoin'] = 'round';
+            }
+
+            if (segment[1]) {
+                attribs.stroke = segment[1];
+            }
+
+            if (isSVGPathSegment(segment[0])) {
+                item = series.chart.renderer.path(segment[0])
+                    .attr(attribs)
+                    .add(series.group);
+            }
+
+            if (item?.shadow) {
+                item.shadow(!!options.shadow);
+            }
+
+            return item;
+        }
+
+        // Draw the graphs
+        let graphs = series.graphs;
+
+        if (graphs.length > 0) { // Cancel running animations, #459
+            // do we have animation
+            graphPaths.forEach(function (segment, j): void {
+                // Update color and path
+                if (series.graphs[j] && isSVGPathSegment(segment[0])) {
+                    series.graphs[j].attr({
+                        d: segment[0],
+                        stroke: segment[1]
+                    });
+                } else {
+                    const formattedSegment = getSegment(segment, colorType);
+
+                    if (formattedSegment) {
+                        series.graphs[j] = formattedSegment;
+                    }
+                }
+            });
+        } else if (graphPaths.length > 0) { // #1487
+            graphs = [];
+            graphPaths.forEach(function (segment, j): void {
+                const formattedSegment = getSegment(segment, colorType);
+
+                if (formattedSegment) {
+                    graphs[j] = formattedSegment;
+                }
+            });
+
+            series.graphs = graphs;
+        }
+        // Checks if series.graphs exists. #3
+        graphSegmentsLength = (series.graphs && series.graphs.length) || -1;
+
+        for (let j = graphSegmentsLength; j >= graphPathLength; j--) {
+            if (series.graphs && series.graphs[j]) {
+                series.graphs[j].destroy();
+                series.graphs.splice(j, 1);
+            }
+        }
+    }
+
+    /**
+     *
+     *  Events
+     *
+     */
+
+    public translate (): void {
+        super.translate.apply(this, arguments);
+
+        if (this.getSegments) {
+            this.getSegments();
+        }
+    }
+}
+
+/**
+ *
+ *  Class Prototype
+ *
+ */
+
+interface ColoredlineSeries extends LineSeries {
+    getPointSpline: (
+        segment: SeriesColoredPoint[],
+        point: SeriesColoredPoint,
+        i: number
+    ) => SeriesColoredSegmentPath[]
+}
+
+extend(ColoredlineSeries.prototype, {
+    getPointSpline: ColoredlineSeries.prototype.getPointSpline
+});
+
```

Comment:
Why we need the above (extend, interface)?

Reply:
We don't define the `getPointSpline` in our package, but as it stands in the code, it should exist on a series, but it's probably not typed correctly in HC. So, this syntax is used to add a missing type and define its existing, actual value.

Reply:
I'm a little confused right now, but after some testing, I agree it's not needed. Thanks!